components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+            LOG.error("Authorize URL creation failed with the additional query parameters issue.", e);
+            throw new UnsupportedEncodingException();
+        }
+        return finalAuthzUrl;
+    }
+
+
+    /**
+     * This method is used to retrieve user claims from user info endpoint.
+     *
+     * @param oAuthResponse
+     * @param context
+     * @return
+     */
+    public Map<String, Object> getUserInfoUserAttributes(OAuthClientResponse oAuthResponse,
+                                                            AuthenticationContext context)
+            throws UAEPassUserInfoFailedException {
+
+        String accessToken = oAuthResponse.getParam(UAEPassAuthenticatorConstants.UAE.ACCESS_TOKEN);
+        String userInfoJsonPayload;
+        Map<String,Object> userInfoJwtAttributes = null;
+
+        try{
+            userInfoJsonPayload = sendUserInfoRequest(context, accessToken);
+            if(LOG.isDebugEnabled()){
+                LOG.debug("Successfully returns the userinfo JSON payload");
+            }
+            Set<Map.Entry<String, Object>> jwtAttributeSet;
+            jwtAttributeSet = JSONObjectUtils.parse(userInfoJsonPayload).entrySet();
+            userInfoJwtAttributes= buildJSON(jwtAttributeSet);
+
+        } catch (UAEPassUserInfoFailedException e){
+                throw new UAEPassUserInfoFailedException("Unable to get retrieve info claims",e);
+        } catch (ParseException e) {
+            if(LOG.isDebugEnabled()) {
+                LOG.error("Error occurred while parsing UserInfo payload by UAEPass: ", e);
+            }
+        }
+        return userInfoJwtAttributes;
+    }
+
+
+    /**
+     * This method is used to create userinfo request with the access token.
+     *
+     * @param context
+     * @param accessToken
+     * @return
+     * @throws IOException
+     */
+    public String sendUserInfoRequest(AuthenticationContext context, String accessToken) throws
+            UAEPassUserInfoFailedException {
+
+        StringBuilder builder = new StringBuilder();
+
+        try {
+            String envUAEPass = getUAEPassEnvironment(context);
+            URL userInfoUrl = new URL(getUserInfoUrl(envUAEPass));
+            HttpURLConnection httpURLConnection = (HttpURLConnection) userInfoUrl.openConnection();
+
+            httpURLConnection.setRequestMethod("GET");
+            httpURLConnection.setRequestProperty("Authorization", "Bearer " + accessToken);
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(httpURLConnection.getInputStream()));
+            String inputLine = reader.readLine();
+
+            if(LOG.isDebugEnabled()){
+                LOG.debug("User info request is sent successfully.");
+            }
+
+            while (inputLine != null) {
+                builder.append(inputLine).append("\n");
+                inputLine = reader.readLine();
+            }
+
+        }catch (IOException e ){
```

Comment:
```suggestion
        } catch (IOException e) {
```
Check formatting in all the places.